[L0] fix windows cmake scripts

pbalcer opened the pull request:
This patch addresses the portability of L0 cmake scripts and adds building L0 adapter in CI on Windows. There were a lot of conversion warnings in the L0 adapter itself, so I disabled these for now. I've also had to exclude windows 2019 from building the adapter because of these errors from level-zero loader:

```
2023-10-03T10:10:14.5073647Z   Building Custom Rule D:/a/unified-runtime/unified-runtime/source/adapters/null/CMakeLists.txt
2023-10-03T10:10:15.0383355Z D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-src\source\loader\windows\driver_discovery_win.cpp(37,69): error C2065: 'GUID_DEVCLASS_COMPUTEACCELERATOR': undeclared identifier [D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-build\source\ze_loader.vcxproj]
2023-10-03T10:10:15.0662878Z D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-src\source\loader\windows\driver_discovery_win.cpp(39,67): error C3536: 'computeDrivers': cannot be used before it is initialized [D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-build\source\ze_loader.vcxproj]
2023-10-03T10:10:15.1634971Z   ur_null.cpp
2023-10-03T10:10:15.2915021Z   ur_nullddi.cpp
2023-10-03T10:10:15.5559593Z D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-src\source\loader\windows\driver_discovery_win.cpp(39,97): error C2661: 'std::vector<loader::DriverLibraryPath,std::allocator<loader::DriverLibraryPath>>::insert': no overloaded function takes 1 arguments [D:\a\unified-runtime\unified-runtime\build\_deps\level-zero-loader-build\source\ze_loader.vcxproj]
```

I tried looking into this (maybe some windows component is missing?), but to no avail unfortunately.

pbalcer commented:
@jsji can you test whether this patch would fix the problem on ICX?

jsji commented:
> @jsji can you test whether this patch fixes the problem on ICX?

Yes, this also fixes the build failures for ICX  -- **it can build successfully despite having warnings emitted.**

However, I got one failure in the CI https://github.com/jsji/unified-runtime/actions/runs/6393683162/job/17353425529

pbalcer commented:
> > @jsji can you test whether this patch fixes the problem on ICX?
> 
> Yes, this also fixes the build failures for ICX -- **it can build successfully despite having warnings emitted.**
> 
> However, I got one failure in the CI https://github.com/jsji/unified-runtime/actions/runs/6393683162/job/17353425529

Great, thanks. I've fixed the CI failure.

I removed disabling the individual warnings because it's something we don't control and keeping track of what warnings to enable/disable in an external dependency might be bothersome (e.g., when updating). Might as well just disable all warnings if we don't want to see them?

lukaszstolarczuk reviewed (commented):
👍
